src/playasophy/wonderdome/mode/strip_eq.clj

```diff
@@ -0,0 +1,75 @@
+(ns playasophy.wonderdome.mode.strip_eq
+  (:require
+    [playasophy.wonderdome.geometry.sphere :as sphere]
+    [playasophy.wonderdome.mode.core :as mode]
+    [playasophy.wonderdome.util.color :as color]
+  )
+)
+
+; StripEQ will process audio signals and turn each strip into an indicator for
+; the strength of a range of frequencies over time.
+(defrecord StripEqMode
+  [num_strips pix_per_strip]
+
+  mode/Mode
+
+  (update
+    [this event]
+    (case (:type event)
+      :time/tick
+      (assoc this
+             ; TODO: Vary colorindex over time smoothly.
+             :colorindex (double (/ 1.0 num_strips))
+      )
+
+      :audio/beat
+      (assoc this
+             ; TODO: Reserve one strip for beats?
+             :beat/at (System/currentTimeMillis)
+             :beat/power (:power event)
+      )
+
+      :audio/freq
+      (assoc this
+             :freq/at (System/currentTimeMillis)
+             ; Break frequencies into groups per strip and take mean for each.
+             :freq/group_mean (map
+               #(/ (reduce + %) (count %))
+               (partition
+                 (int (/ (count (:spectrum event)) num_strips))
+                 (:spectrum event)
+               )
+             )
```

Comment:
This section could also use some cleanup. Typically you'd align things to the first argument to a function if you break it over multiple lines, or indented two from the function name if all the arguments are on separate lines. Threading macros also help a lot when you have deeply-nested forms:

```clojure
(->> (:spectrum event)
     (partition (-> event :spectrum count (/ num-strips) int)
     (map #(/ (reduce + %) (count %))))
```